src/OrchardCore.Modules/OrchardCore.Features/Controllers/AdminController.cs

```diff
@@ -58,11 +62,14 @@ public async Task<ActionResult> Features(string tenant)
 
             var viewModel = new FeaturesViewModel();
 
+            var isDefaultTenant = _shellSettings.IsDefaultShell();
+
             await ExecuteAsync(tenant, async (featureService, settings) =>
             {
                 // if the user provide an invalid tenant value, we'll set it to null so it's not available on the next request
                 tenant = settings?.Name;
                 viewModel.Name = settings?.Name;
+                viewModel.IsDefaultTenant = isDefaultTenant && (settings == null || settings.IsDefaultShell());
```

Comment:
What you could do is to always pass a non null settings in your delegate even if you didn't create a tenant scope (acting on the current tenant), this would simplify the check => `isDefaultTenant && settings.IsDefaultShell()`

Replacing

    // at this point we apply the action on the current tenant
     await action(new FeatureService(_shellFeaturesManager, _extensionManager), null);

with

    await action(new FeatureService(_shellFeaturesManager, _extensionManager), _shellSettings);

Also under `shellScope.UsingAsync()`.

    // at this point we apply the action on the given tenant
    await action(new FeatureService(shellFeatureManager, extensionManager), settings);

We could do

    await action(new FeatureService(shellFeatureManager, extensionManager), scope.ShellContext.Settings);

Reply:
Sure. I'll make these changes tomorrow. But I we need to agree on the other item

Reply:
I made the requested change.